release/tar/linux/opensearch-tar-build.sh

```diff
@@ -93,6 +94,13 @@ then
     exit 1
 fi
 
+if [[ ( "$ARCHITECTURE" == "arm64" && ( "$UNDERLYING_ARCH" != "aarch64" && "$UNDERLYING_ARCH" != "arm64" )) ||  ( "$ARCHITECTURE" == "x64" && ( "$UNDERLYING_ARCH" != "x86_64" && "$UNDERLYING_ARCH" != "amd64" )) ]]
```

Comment:
You mean `( "$UNDERLYING_ARCH" != "aarch64" || "$UNDERLYING_ARCH" != "arm64" )`?
Since a server cannot be both aarch64 and arm64 at the same time.

Reply:
Actually confused a bit why it is a problem with underlying arch.
If user building arm64 on x64 and failed I can get it.
However, build arm64 on aarch64 is not a concern since they are internally the same arch, just name differences.

Reply:
yes aarch64 and arm64 for building arm is okay. That is what this logic does. Tested it on arm64 and x64. You can try too. If manifest contains arch as `arm64` and underlying system is neither aarch64 `&&` arm64 then throw this error